test/test-lib/fixtures.ts

```diff
@@ -0,0 +1,697 @@
+	}
+
+	await awaitFixtures(fixtures);
+	const [response, body] = await requestAsync(requestOpts);
+
+	if (response.statusCode !== 201) {
+		logErrorAndThrow('Failed to create application', jsonData);
+	}
+	return body;
+};
+
+loaders['configs'] = jsonData => {
+	const { resinApi, root } = loadPlatform().platform;
+	return resinApi.post({
+		resource: 'config',
+		body: _.pick(jsonData, 'key', 'value', 'scope', 'description'),
+		passthrough: { req: root },
+	});
+};
+
+loaders['device-environment-variables'] = async (jsonData, fixtures) => {
+	const device = await fixtures.devices[jsonData.device];
+	if (device == null) {
+		logErrorAndThrow(`Could not find device: ${jsonData.device}`);
+	}
+	const user = await fixtures.users['root'];
+
+	const requestOpts = buildApiRequestOpts({
+		path: 'resin/device_environment_variable',
+		body: {
+			device: device.id,
+			name: jsonData.name,
+			value: jsonData.value,
+		},
+		user,
+	});
+
+	// We depend on `builds` and `service_installs` as device env vars depend on service installs (when converted to service env vars)
+	// and those are the two fixtures that can create them
+	await awaitFixtures(fixtures, 'builds', 'service_installs');
+
+	const [response, body] = await requestAsync(requestOpts);
+
+	if (response.statusCode !== 201) {
+		logErrorAndThrow('Failed to create device environment variable', jsonData);
+	}
+	return body;
+};
+
+loaders['device-tags'] = async (jsonData, fixtures) => {
+	const device = await fixtures.devices[jsonData.device];
+	if (device == null) {
+		logErrorAndThrow(`Could not find device: ${jsonData.device}`);
+	}
+	const user = await fixtures.users['root'];
+
+	const requestOpts = buildApiRequestOpts({
+		path: 'resin/device_tag',
+		body: {
+			device: device.id,
+			tag_key: jsonData.tag_key,
+			value: jsonData.value,
+		},
+		user,
+	});
+	const [response, body] = await requestAsync(requestOpts);
+
+	if (response.statusCode !== 201) {
+		logErrorAndThrow('Failed to create device tag', jsonData);
+	}
+	return body;
+};
+
+loaders['devices'] = async (jsonData, fixtures) => {
+	const application = await fixtures.applications[
+		jsonData.belongs_to__application
+	];
+	if (application == null) {
+		logErrorAndThrow(
+			`Could not find application: ${jsonData.belongs_to__application}`,
+		);
+	}
+	const user = await fixtures.users['root'];
+
+	const requestOpts = buildApiRequestOpts({
+		path: 'resin/device',
+		body: {
+			belongs_to__application: application.id,
+			belongs_to__user: user.id,
+			..._.pick(
+				jsonData,
+				'uuid',
+				'device_type',
+				'is_managed_by__device',
+				'os_version',
+				'supervisor_version',
+				'logs_channel',
+			),
+		},
+		user,
+	});
+	const [response, body] = await requestAsync(requestOpts);
+
+	if (response.statusCode !== 201) {
+		logErrorAndThrow('Failed to create device', jsonData, body);
+	}
+	return body;
+};
+
+loaders['image_environment_variables'] = async (jsonData, fixtures) => {
```

Comment:
Some loaders are keyed by names that use dashes `-`, others with underscores `_`. I think it's worth picking one of the two styles now that no fixtures exist yet.

Reply:
agreed - I will go with whatever we use in Pine resource queries, since dashes won't work directly anyway (`obj.a-b-c` is invalid) and underscores will trigger TSLint (`obj.a_b_c` will be snake-case) so in both instances we would refer via index (`obj['a-b-c']` or `obj['a_b_c']`)

Reply:
I will do the `a_b_c` as this is how we reference the resources in Pine. Keeps it aligned 👍